File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/PlayNextSong.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState;

import java.util.List;
import java.util.Optional;

public record PlayNextSong(
        @Json(name = "songNumber")
        int songNumber,
        @Json(name = "onLastSong")
        boolean onLastSong,
        @Json(name = "multipleChoiceNames")
        Optional<List<String>> multipleChoiceNames,
        @Json(name = "time")
        double time,
        @Json(name = "extraGuessTime")
        int extraGuessTime,
        @Json(name = "progressBarState")
        ProgressBarState progressBarState
) implements Command {

    @Override
    public String commandName() {
        return CommandTypeOld.PLAY_NEXT_SONG.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/PlayNextSong.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState

@JvmRecord
@CommandType("play next song")
data class PlayNextSong(
    @Json(name = "songNumber")
    val songNumber: Int,
    @Json(name = "onLastSong")
    val onLastSong: Boolean,
    @Json(name = "multipleChoiceNames")
    val multipleChoiceNames: List<String>?,
    @Json(name = "time")
    val time: Double,
    @Json(name = "extraGuessTime")
    val extraGuessTime: Int,
    @Json(name = "progressBarState")
    val progressBarState: ProgressBarState
)
